rubrics/tool/src/main/java/org/sakaiproject/rubrics/controller/GeneratePdfController.java

```diff
@@ -0,0 +1,268 @@
+import org.sakaiproject.rubrics.logic.model.Criterion;
+import org.sakaiproject.rubrics.logic.model.CriterionOutcome;
+import org.sakaiproject.rubrics.logic.model.Evaluation;
+import org.sakaiproject.rubrics.logic.model.Rating;
+import org.sakaiproject.rubrics.logic.model.Rubric;
+import org.sakaiproject.rubrics.logic.repository.EvaluationRepository;
+import org.sakaiproject.rubrics.logic.repository.RubricRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.data.repository.query.Param;
+import org.springframework.data.rest.webmvc.RepositoryRestController;
+import org.springframework.data.rest.webmvc.support.RepositoryEntityLinks;
+import org.springframework.http.ContentDisposition;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody;
+
+import com.itextpdf.text.BaseColor;
+import com.itextpdf.text.Chunk;
+import com.itextpdf.text.Document;
+import com.itextpdf.text.DocumentException;
+import com.itextpdf.text.Element;
+import com.itextpdf.text.Font;
+import com.itextpdf.text.PageSize;
+import com.itextpdf.text.Paragraph;
+import com.itextpdf.text.Phrase;
+import com.itextpdf.text.pdf.PdfPCell;
+import com.itextpdf.text.pdf.PdfPTable;
+import com.itextpdf.text.pdf.PdfWriter;
+
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.user.api.PreferencesService;
+
+import org.jsoup.Jsoup;
+
+@Slf4j
+@RepositoryRestController
+@NoArgsConstructor
+@AllArgsConstructor
+public class GeneratePdfController {
+
+    @Autowired
+    RubricsConfiguration rubricsConfiguration;
+
+    @Autowired
+    RepositoryEntityLinks repositoryEntityLinks;
+    
+    @Autowired
+    private MessageSource messageSource;
+    
+    @Autowired
+    private SiteService siteService;
+    
+    @Autowired
+    private RubricRepository rubricRepository;
+    
+    @Autowired
+    private EvaluationRepository evaluationRepository;
+    
+    @Autowired
+    private SessionManager sessionManager;
+
+    @Autowired
+    private PreferencesService preferencesService;
+
+    private static final String X_COPY_HEADER = "x-copy-source";
+    private static final String X_LANG_HEADER = "lang";
+    private static Font font = new Font(Font.FontFamily.HELVETICA, 10, Font.BOLD);
+    private static Font textFont = new Font(Font.FontFamily.HELVETICA, 8, Font.BOLD);
+    private static Font descFont = new Font(Font.FontFamily.HELVETICA, 7, Font.NORMAL);
+
+    @PreAuthorize("canCopy(#sourceId, 'Rubric')")
```

Comment:
I don't see why it should be bound to that permission, what do you think @adrianfish ?